### PR TITLE
Feature edit profile dialog #25

### PR DESCRIPTION
**Pull request by tmclean15**

### CODE REVIEW

File: client/src/components/ExpertiseChips.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      else return [...chips, expertise];
+    });
+    setExpertise("");
+  };
+
+  const handleRemoveChip = (chip) => {
+    setChips((chips) => {
+      const index = chips.indexOf(chip);
+      chips.splice(index, 1);
+      return [...chips];
```
Comment (bonnieli) [inline client/src/components/ExpertiseChips.js]:
```suggestion
      return chips.filter((c) => return c !== chip)
```

i _think_ that should be a similar and shorter way to filter out that specific chip :)

Comment (tmclean15) [reply client/src/components/ExpertiseChips.js]: Much cleaner, thanks!

File: server/api/user_handler.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
 user_handler = Blueprint('user_handler', __name__)
 
 
-@user_handler.route('/api/v1/users/<user_id>/profile', methods=['GET'])
+@user_handler.route('/api/v1/user/<user_id>/profile', methods=['GET'])
```
Comment (MalcolmAG) [inline server/api/user_handler.py:10]: I think we should keep this as '/api/v1/users/<user_id>/profile'

Comment (MalcolmAG) [inline server/api/user_handler.py]: I think we should keep this as '/api/v1/users/<user_id>/profile'

File: server/api/user_handler.py [hunk 3]
```diff
@@ -18,7 +18,7 @@ def get_user(user_id):
     return jsonify(user.serialize), 200
 
 
-@user_handler.route('/api/v1/users/<user_id>/profile', methods=['PUT'])
+@user_handler.route('/api/v1/user/<user_id>/profile', methods=['PUT'])
```
Comment (MalcolmAG) [inline server/api/user_handler.py]: Same here

File: client/src/components/DropZoneUpload.js [hunk 4]
```diff
@@ -1,9 +1,14 @@
...
 import { DropzoneDialog } from "material-ui-dropzone";
 import axios from "axios";
 
-function DropZoneUpload(props) {
+function DropZoneUpload({
+  uploadLocation,
+  projectId,
+  upload,
+  handleUploadSuccess,
+}) {
```
Comment (MalcolmAG) [inline client/src/components/DropZoneUpload.js:11]: A small question: Is this the preferred method with components?

File: client/src/components/EditProfileDialog.js [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+  };
+
+  const handleTriggerFileUpload = (event) => {
+    // Trigger photo upload to s3
+    setUpload(true);
+
+    setDialogOpen(false);
+  };
+
+  const handleSave = useCallback((profilePics) => {
```
Comment (MalcolmAG) [inline client/src/components/EditProfileDialog.js]: A small question, is "useCallback" necessary here? There's no dependencies, so it acts just like a function.

**Review by MalcolmAG — CHANGES_REQUESTED** [review]
Just a small request for the backend user routes, and other small questions throughout.

File: server/api/user_handler.py (hunk 2 above)
Comment (tmclean15) [reply server/api/user_handler.py]: I was looking to make it consistent with the project routes, but I do think plural is better. I'll change them back.

File: client/src/components/DropZoneUpload.js (hunk 4 above)
Comment (tmclean15) [reply client/src/components/DropZoneUpload.js:11]: Generally yes, it avoids needing to use `props.projectId`, `props.upload`, etc.

File: client/src/components/EditProfileDialog.js (hunk 5 above)
Comment (tmclean15) [reply client/src/components/EditProfileDialog.js]: The function is actually dependent on the state of the EditProfileDialog component, I just forgot to pass them in as dependencies. I will add those in.

**Review by MalcolmAG — APPROVED** [review]
Looks good

**Pull request merged**